x-pack/plugins/security_solution/public/entity_analytics/components/risk_engine_privileges_callout/risk_engine_privileges_callout.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import hash from 'object-hash';
+import type { CallOutMessage } from '../../../common/components/callouts';
+import { CallOutSwitcher } from '../../../common/components/callouts';
+import { MissingPrivilegesCallOutBody, MISSING_PRIVILEGES_CALLOUT_TITLE } from './translations';
+import { useMissingPrivileges } from './use_missing_risk_engine_privileges';
+
+export const RiskEnginePrivilegesCallOut = () => {
+  const missingPrivileges = useMissingPrivileges();
+
+  const MissingPrivilegesMessage: CallOutMessage | null = useMemo(() => {
+    const hasMissingPrivileges =
+      missingPrivileges.indexPrivileges.length > 0 ||
+      missingPrivileges.clusterPrivileges.length > 0 ||
+      missingPrivileges.kibanaPrivileges.length > 0;
+
+    if (!hasMissingPrivileges) {
+      return null;
+    }
+
+    const missingPrivilegesHash = hash(missingPrivileges);
+    return {
+      type: 'primary',
+      /**
+       * Use privileges hash as a part of the message id.
+       * We want to make sure that the user will see the
+       * callout message in case his privileges change.
+       * The previous click on Dismiss should not affect that.
+       */
+      id: `missing-risk-engine-privileges-${missingPrivilegesHash}`,
```

Comment:
Have this behavior been reproduced? For the logic of this component, it looks like it will work without this hash. When the `missingPrivileges` array changes this useMemo is re-executed, then a new `missingPrivilegesMessage` object is returned, so the `CallOutSwitcher` should update with the new message, without the need of having a different `id`. Am I missing something?

Reply:
Confirmed we dont need and removed in https://github.com/elastic/kibana/pull/171250/commits/0435e57531ff4bfeb33ac902e5a12b25ac1295c2